kedro/config/omegaconf_config.py

```diff
@@ -335,7 +335,8 @@ def load_and_merge_dir_config(  # noqa: PLR0913
                 ) from exc
 
         seen_file_to_keys = {
-            file: set(config.keys()) for file, config in config_per_file.items()
+            file: self._get_all_keys(OmegaConf.to_container(config, resolve=False))
```

Comment:
I think the case of having the same level top-key is only realistic for `parameters` and not the `catalog`, so I'd change this to only call `_get_all_keys` for parameters.

Reply:
Makes sense. Updated the code.